compiler/crates/relay-config/src/config-schema.json

```diff
@@ -0,0 +1,588 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
```

Comment:
Do you think we could add a comment above each struct/enum that is included here informing folks that changes made to that struct need to be replicated here?

Reply:
This is also making me think that for folks who describe their config using typescript, it would be nice to have typescript definitions as well. Are there tools to generate typescript from json-schema (or vise-versa?)

Reply:
I've used a JSON Schema to TypeScript npm package before, but I'm sure it exists the other way around as well. 
So you're saying there should be TypeScript types for the config. 
Where should these live? `@types/relay-compiler`? I think an entirely different repository will be even harder to sync. I'm personally of the opinion that the types should be inclined into the Relay repository, so I could also just add it straight to the relay-compiler package. We could then have a script that created the JSON schema from these TypeScript definitions at build, so that only the TS types have to be kept in sync.

Reply:
Okay I've added the TypeScript types and tested two packages to generate the JSON schema from the TypeScript types. In both cases the schema is losing expressiveness in the form `oneOf`. With the handwritten schema you could for example express that only "schema" or "schemaDir" is supported, which is now no longer possible. If we go the reverse and generate the Types from the JSON schema, the types are also weird in some places.
Is this loss of expressiveness okay in light of the easier maintainability? I guess it's still better to have some guidance, but it could also be confusing because both Single project and Multiproject options are available at the root, which would lead to the compiler panicking...

Reply:
What's the reason for needing two scripts to generate the schema? I don't follow that bit.

I suspect the best fix for the oneOf issue would be to move to a config format that was less overloaded.

What do you personally think of the tradeoff of manual vs generated? It seems like the generated one is still much better than nothing, even if it's not optimal, and that it's better than having to maintain two different versions, but I haven't tried it out enough to know how badly the generated schema misses the mark.

Reply:
It's just to showcase what two different packages would output. This is not final. The generated JSON file would also not be checked in normally and the build integration is still missing. I'll mark as draft again to avoid confusion.

I would prefer the generated, if it was a bit better.
The goal of this was to assist you in writing the config file. It does that, but the oneOf issue suggested invalid configuration options that might lead to errors, which is bad. I'll try to see if there is something I can do to improve the situation, but I don't have anymore time today, sadly...